### PR TITLE
scourge, new recipe

### PR DESCRIPTION
**Pull request by Begasus**
Marked as ? as I can't seem to run the game:
```
 ~> scourge 
  Constructing root path:
          not using binreloc...
          temp rootDir=/packages/scourge_x86-0.21.1-1/.self/data/scourge
  *** Can't find local version of data dir. You're running a distribution.
  Looking for localized resources in: /packages/scourge_x86-0.21.1-1/.self/data/scourge/translations
  Starting session. Final rootDir=/packages/scourge_x86-0.21.1-1/.self/data/scourge
  Setting video mode: 1024x768x32
  GalliumContext: CreateScreen: Using llvmpipe (LLVM 5.0, 128 bits) driver.
  Unlocking BLocker with sem 1841914 from wrong thread 65333, current holder -1 (see issue #6400).
```

### CODE REVIEW

**Comment by Begasus** [comment]
Based on the original bep files from @michaelvoliveira

File: games-roguelike/scourge/patches/scourge-0.21.1.patchset [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+diff --git a/src/equip.cpp b/src/equip.cpp
+index 2c74b82..351536f 100644
+--- a/src/equip.cpp
++++ b/src/equip.cpp
+@@ -595,7 +595,7 @@ void Equip::drawCapabilities() {
+ 		}
+ 	}
+ 	if ( !found ) {
+-		specialSkill = false;
++		specialSkill = FALSE;
```
Comment (alaviss) [inline games-roguelike/scourge/patches/scourge-0.21.1.patchset]:
```suggestion
+		specialSkill = NULL;
```

File: games-roguelike/scourge/patches/scourge-0.21.1.patchset [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+diff --git a/src/render/map.cpp b/src/render/map.cpp
+index 9a80e73..6955224 100644
+--- a/src/render/map.cpp
++++ b/src/render/map.cpp
+@@ -3154,7 +3154,7 @@ Shape *Map::isWallBetween( int x1, int y1, int z1, int x2, int y2, int z2 ) {
+ 			if ( shape )
+ 				return shape;
+ 		}
+-		return false;
++		return FALSE;
```
Comment (alaviss) [inline games-roguelike/scourge/patches/scourge-0.21.1.patchset]:
```suggestion
+		return NULL;
```

File: games-roguelike/scourge/patches/scourge-0.21.1.patchset [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+ 	}
+ 	if ( y1 == y2 ) {
+ 		if ( x1 > x2 ) SWAP( x1, x2 );
+@@ -3163,7 +3163,7 @@ Shape *Map::isWallBetween( int x1, int y1, int z1, int x2, int y2, int z2 ) {
+ 			if ( shape )
+ 				return shape;
+ 		}
+-		return false;
++		return FALSE;
```
Comment (alaviss) [inline games-roguelike/scourge/patches/scourge-0.21.1.patchset]:
```suggestion
+		return NULL;
```

**Comment by Begasus** [comment]
This is the error from config.log without patching configure.in (for glX):
```
configure:12172: checking for GL/gl.h
configure:12178: gcc -E    -DHAVE_SDL=1 -DHAVE_SDL_MIXER=1 -DHAVE_SDL_NET=1 -I/packages/freetype_x86-2.9-1/.self/develop/headers/x86/freetype2 conftest.c
configure:12178: $? = 0
configure:12184: result: yes
configure:12244: checking GL/glext.h usability
configure:12244: gcc -c -g -O2    -DHAVE_SDL=1 -DHAVE_SDL_MIXER=1 -DHAVE_SDL_NET=1 -I/packages/freetype_x86-2.9-1/.self/develop/headers/x86/freetype2 conftest.c >&5
In file included from conftest.c:131:0:
/boot/system/develop/headers/x86/GL/glext.h:111:54: error: unknown type name 'GLenum'; did you mean 'enum'?
 typedef void (APIENTRYP PFNGLDRAWRANGEELEMENTSPROC) (GLenum mode, GLuint start, GLuint end, GLsizei count, GLenum type, const void *indices);
                                                      ^~~~~~
                                                      enum
/boot/system/develop/headers/x86/GL/glext.h:111:67: error: unknown type name 'GLuint'; did you mean 'uint'?
 typedef void (APIENTRYP PFNGLDRAWRANGEELEMENTSPROC) (GLenum mode, GLuint start, GLuint end, GLsizei count, GLenum type, const void *indices);
                                                                   ^~~~~~
                                                                   uint
/boot/system/develop/headers/x86/GL/glext.h:111:81: error: unknown type name 'GLuint'; did you mean 'uint'?
 typedef void (APIENTRYP PFNGLDRAWRANGEELEMENTSPROC) (GLenum mode, GLuint start, GLuint end, GLsizei count, GLenum type, const void *indices);
                                                                                 ^~~~~~
                                                                                 uint
/boot/system/develop/headers/x86/GL/glext.h:111:93: error: unknown type name 'GLsizei'
 typedef void (APIENTRYP PFNGLDRAWRANGEELEMENTSPROC) (GLenum mode, GLuint start, GLuint end, GLsizei count, GLenum type, const void *indices);
                                                                                             ^~~~~~~
```

File: games-roguelike/scourge/patches/scourge-0.21.1.patchset (hunk 3 above)
Comment (Begasus) [reply games-roguelike/scourge/patches/scourge-0.21.1.patchset]: bumped these locally already, will push those later :+1:

**Comment by Begasus** [comment]
Terminal output without patching for glX (just removed the tests for glext.h):
```
~> scourge 
Constructing root path:
        not using binreloc...
        temp rootDir=/packages/scourge_x86-0.21.1-1/.self/data/scourge
*** Can't find local version of data dir. You're running a distribution.
Looking for localized resources in: /packages/scourge_x86-0.21.1-1/.self/data/scourge/translations
Starting session. Final rootDir=/packages/scourge_x86-0.21.1-1/.self/data/scourge
Setting video mode: 800x640x32
GalliumContext: CreateScreen: Using llvmpipe (LLVM 5.0, 128 bits) driver.
Unlocking BLocker with sem 109768 from wrong thread 12789, current holder -1 (see issue #6400).
For font >normal< using locale >all<
For font >ui< using locale >all<
For font >mono< using locale >all<
For font >large< using locale >all<
*** Error: couldn't generate display list for gui.
```

**Comment by Begasus** [comment]
Fixed path to the data-dir, fixed reference to lib:libSDL2_2.0 (related to https://github.com/haikuports/haikuports/issues/3878)

**Pull request merged**